src/client/service-worker/index.js

```diff
@@ -15,13 +15,20 @@
  */
 
 export function importServiceWorkerClient(settings) {
-  // In the future, we should consider making one-service worker added dynamically
-  // as an external chunk with the ability to support conditional load.
   return import(/* webpackChunkName: "service-worker-client" */ './client')
+    // get the default export
     .then((imported) => imported.default)
+    // invoke with settings
     .then((serviceWorkerClient) => serviceWorkerClient(settings));
```

Comment:
Why not remove the extra `then` handler?
```suggestion
return import(/* webpackChunkName: "service-worker-client" */ './client')
    // get the default export and invoke with settings
    .then(({ default: serviceWorkerClient }) => serviceWorkerClient(settings)
```